miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleController.java

```diff
@@ -218,6 +199,20 @@ public SampleDto createSample(@RequestBody SampleDto sampleDto, UriComponentsBui
     return created;
   }
 
+  private <T extends DetailedSampleDto> Long inferIntermediateSampleClassId(T dto, Function<? super T, Long> getId, String currentCategory,
```

Comment:
Please rename parameters for clarity or add Javadoc-style comment to explain
* getId = getChildClassId (or just accept childClassId instead of Function)
* currentCategory = childClassCategory